core/modules/system/system.theme.inc

```diff
@@ -103,12 +103,29 @@ function theme_system_themes_page($variables) {
 
       // Localize the theme description.
       $description = t($theme->info['description']);
+      $prefix = '';
+      $depends = '';
+      if (isset($theme->info['base theme'])) {
+        $base_theme_name = $theme->info['base theme'];
+        $base_theme_label = !empty($theme->base_themes[$base_theme_name]) ? $theme->base_themes[$base_theme_name] : $base_theme_name;
+        $depends = '<br><strong>' . t('Base theme') . ':</strong> ' . check_plain($base_theme_label);
+        if (isset($theme->missing_dependencies)) {
+          $theme->classes[] = 'missing-dependency';
+          $icon = icon('warning', array(
+            'attributes' => array('width' => 24, 'height' => 24),
+          ));
+          $prefix = '<div class="prefix">' . $icon . ' ' . t('Missing dependency') . '</div>';
```

Comment:
```suggestion
          $prefix = '<div class="prefix">' . $icon . ' ' . t('Missing dependency') . '</div>';
```

Is there a particular reason why the variable and class is `prefix`?  I understand there is a `missing-dependency` class on the wrapper around each theme's description but would something like `warning` or `missing-warning` perhaps be a more meaningful class and variable for this part of it as this is all it is used for?

In the modules page, if a module has missing requirements, then the class around the "missing" text is `admin-missing`

Reply:
It's `$prefix` as it's usually just an empty string, not necessarily a warning.

The CSS class might be better renamed to something "more speaking", I think, you're right re that. :+1:
Both CSS classes currently aren't actually used by Seven, but I thought it might make sense to add for easier styling (if an admin theme wants to).